src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/References/SetTreatAsUsedAttributeCommand.cs

```diff
@@ -32,7 +32,7 @@ public async Task<bool> ApplyAsync(CancellationToken cancellationToken)
                 return false;
             }
 
-            await item.Metadata.SetPropertyValueAsync(ProjectReference.TreatAsUsedProperty, SetValue);
+            await item.Metadata.SetPropertyValueAsync(ProjectReference.TreatAsUsedProperty, SetTreatAsUsed);
```

Comment:
Feel like these Apply and Revert methods should invoke abstract methods that each subclass would implement. That way they can SetTreatAsUsed and UnsetTreatAsUsed appropriately.

Reply:
I say that because the SetAttributeCommand and UnusetAttributeCommand are now identical and would perform the save action on Apply and Revert.